### DISCUSSION TITLE
Rule for no `text ""`

### DISCUSSION THREAD

**Discussion by leojpod** [jfmengels/elm-review#68] [Rule ideas]
**What the rule should do:**
This rule should flag the use of `Html.text` with an empty string literal and suggest to use one of the alternative from `elm-community/html-extra` instead.

**Example of things the rule would report:**

```elm
stuff |> Maybe.map viewStuff |> Maybe.withDefault (text "")
```

**Example of things the rule would not report:**

```elm
import ModuleWithText exposing (text)

stuff |> Maybe.map viewStuff |> Maybe.withDefault (text "")
```

**When (not) to enable this rule:**

- if you do not want to use `Html.Extra`
- if you do not mind using `text ""`

**I am looking for:**

- Feedback
- To know in which package(s) this would fit best or if you've other rules that would fit in. 

At the moment the package is published here: https://package.elm-lang.org/packages/leojpod/review-no-empty-html-text/latest/

**Comment by MartinSStewart** [comment]
I'm not sure if I understand the difference is between the example that the rule reports and the example the rule doesn't report. Is it that that the module has WithText in the name?

**Comment by leojpod** [comment]
No :)

The difference is that the rule should target only `Html.text` and not a function `text` from another module or even a function that is define in the file.

**Comment by MartinSStewart** [comment]
Oh, yes in hindsight that should have been obvious. Thanks for the clarification.

**Comment by leojpod** [comment]
No problems at all :) I'm not known for being very clear in my explanations
